color_balance/colorimage.py

```diff
@@ -89,95 +71,67 @@ def convert_to_colorimage(cimage, band_indices=None,
     # Scale the band intensities and mask out-of-range pixels
     out_bands = []
     for band in bands:
-        # If 16-bit, convert to 8-bit, assuming 16-bit image is using the
-        # entire bit depth unless bit depth is given
-        if band.dtype == numpy.uint16:
-            if bit_depth is not None:
-                band = (band / (bit_depth / 2 ** 8))
-            else:
-                band = (band / 2 ** 8)
-        elif band.dtype != numpy.uint8:
-            raise ImagePropertyException(
-                "Band type is {}, should be {} or {}".
-                format(band.dtype,
-                       numpy.uint8, numpy.uint16))
-
-        # Mask out values above or below the 8-bit extents
-        # This is only necessary if bit depth is given
-        clip_max = band > 255
-        cmask[clip_max] = 0
-        clip_min = band < 0
-        cmask[clip_min] = 0
-        out_bands.append(band.astype(numpy.uint8))
-
-    cimg = rgb_bands_to_colorimage(out_bands)
-    return cimg, cmask
-
-
-def rgb_bands_to_colorimage(rgb_bands):
-    return cv2.merge(list(reversed(rgb_bands)))
+        out_bands.append(band)
 
+    # Mask out values above or below the 8-bit extents
+    # This is only necessary if bit depth is given
+    clip_max = band > 255
+    cmask[clip_max] = 0
+    clip_min = band < 0
+    cmask[clip_min] = 0
 
-def colorimage_to_rgb_bands(c_img):
-    return reversed(cv2.split(c_img))
+    cimg = np.dstack(out_bands[::-1])
+    return cimg, cmask
 
 
 def get_histogram(band, mask=None):
-    '''Calculate the histogram of a band. If a mask is provided, masked pixels
-    are not considered in the histogram calculation.
+    """
+    Calculate the histogram of a band. If a mask is provided, masked pixels
+    are not considered in the histogram calculation. The band parameter
+    is assumed to represent a 12bit image when the dtype is np.uint16.
 
-    :param numpy array band: source for calculating histogram
-    :param numpy array mask: array containing zero values at locations where
-        pixels should not be considered, 255 elsewhere
-    :returns: count of pixels at each possible intensity value
-    '''
-    if band.max() > 255 or band.min() < 0:
-        raise OutOfRangeException("Band values outside of [0, 255]")
+    :param band: ndarray of the source for calculating histogram
```

Comment:
Does removal of the type information improve docs formatting?  Where practical I've tried to use try types in :param: directives like ":param int band_num:" but perhaps none is better than not-real parameter types?